### PR TITLE
filesRequired is optional for uploads

### PR DESCRIPTION
**Pull request by TomDo1234**
## Description

Discussed with Jmikrut

- [X ] I have read and understand the [CONTRIBUTING.md](https://github.com/payloadcms/payload/blob/main/CONTRIBUTING.md) document in this repository.

## Type of change

- [ ] Chore (non-breaking change which does not add functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Change to the [templates](../templates/) directory (does not affect core functionality)
- [ ] Change to the [examples](../examples/) directory (does not affect core functionality)
- [X] This change requires a documentation update

## Checklist:

- [ X] I have added tests that prove my fix is effective or that my feature works
- [X] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

**Comment by TomDo1234** [comment]
@jmikrut 

^

File: packages/payload/src/collections/operations/create.ts [hunk 1]
```diff
@@ -128,7 +128,7 @@ async function create<TSlug extends keyof GeneratedTypes['collections']>(
       data,
       overwriteExistingFiles,
       req,
-      throwOnMissingFile: !shouldSaveDraft,
+      throwOnMissingFile: !shouldSaveDraft && collection.config.upload.filesRequiredOnCreate,
```
Comment (jmikrut) [inline packages/payload/src/collections/operations/create.ts:131]: This should be a bit more explicit. Right now, this is written as though you would manually NEED to define `upload.filesRequiredOnCreate: true` for file presence to be enforced.

Rather, we should write something like: 

```
throwOnMissingFile: !shouldSaveDraft && collection.config.upload.filesRequiredOnCreate !== false,
```

**Review by jmikrut — CHANGES_REQUESTED** [review]
Just a quick thing I caught, but otherwise, I'm good with this if you add the property to the docs as well, here:

https://github.com/payloadcms/payload/blob/main/docs/upload/overview.mdx#L43

Then I will happily merge this!

File: packages/payload/src/collections/operations/create.ts (hunk 1 above)
Comment (TomDo1234) [reply packages/payload/src/collections/operations/create.ts]: All done Jmi, all done.

**Pull request merged**